functions/__fzf_search_shell_variables.fish

```diff
@@ -29,21 +29,23 @@ function __fzf_search_shell_variables --argument-names set_show_output set_names
     # with a $, remove it from the query so that it will better match the variable names
     set cleaned_curr_token (string replace -- '$' '' $current_token)
 
-    set variable_name (
+    set variable_names_selected (
         printf '%s\n' $all_variable_names |
-        fzf --preview "__fzf_extract_var_info {} $set_show_output" \
+        fzf --multi --preview "__fzf_extract_var_info {} $set_show_output" \
             --query=$cleaned_curr_token \
             $fzf_shell_vars_opts
     )
 
     if test $status -eq 0
         # If the current token begins with a $, do not overwrite the $ when
         # replacing the current token with the selected variable.
-        if string match --quiet -- '$*' $current_token
-            commandline --current-token --replace \$$variable_name
-        else
-            commandline --current-token --replace $variable_name
-        end
+        commandline --current-token --replace (
```

Comment:
We can set a variable in the `if` and `commandline --current-token --replace $variable` instead, but I found this method more readable, and personally speaking, more beautiful.

Reply:
can you clarify what the other option you had in mind is?

Reply:
oh I get it, you mean set the variable to be written to the commandline in the if block. yeah I like this way better as well